allow sentinel replication configuration

othillo opened the pull request:
I would like to use this bundle with a sentinel setup. This pull request allows setting the replication option to 'sentinel' and allows setting the 'service' option.

related to https://github.com/snc/SncRedisBundle/pull/274

MyDigitalLife commented:
I checked out this code and it works as i was hopping in my test environment. I did however add a extra parameter to make sure selecting a database is an option.

MyDigitalLife commented on DependencyInjection/Configuration/Configuration.php:
```diff
@@ -135,7 +135,8 @@ private function addClientsSection(ArrayNodeDefinition $rootNode)
                                     ->scalarNode('prefix')->defaultNull()->end()
-                                    ->booleanNode('replication')->defaultFalse()->end()
+                                    ->enumNode('replication')->values(array(true, false, 'sentinel'))->defaultNull()->end()
+                                    ->scalarNode('service')->defaultNull()->end()
```
Adding this will enable people to select a database if they choose:
` ->arrayNode('parameters')
      ->children()
         ->scalarNode('database')->defaultNull()->end()
       ->end()
 ->end()`

othillo replied:
@MyDigitalLife do you mind creating a pull request for this with a test case?

othillo commented:
I removed the defaultNull for replication options as suggested by https://github.com/ivoba in https://github.com/snc/SncRedisBundle/pull/305

Also added a test for the cluster configuration where the replication options must not be set.

snc commented:
What does `service: mymaster` is for? I think it would be nice to add something to the docs about sentinal support.

MyDigitalLife commented:
@snc `service: mymaster` is needed to specify a set of instances. This is done within the sentinel config file. You can find some more information about this under "Configuring Sentinel" here: https://redis.io/topics/sentinel

ibrambe commented:
I'm also interested in this feature.

ricbra commented:
Any update on why this isn't merged yet?

snc commented:
@othillo Could you add some lines to the readme about this feature?

othillo commented:
@snc done :) https://github.com/snc/SncRedisBundle/pull/307/commits/b61f0f52b3a79d9dbdd8aa159185f923758fed28

snc commented on Resources/doc/index.md:
````diff
@@ -120,6 +120,24 @@ snc_redis:
+                service: mymaster
+```
+
+The `service` is the name of the set or Redis instances.
````
Should be "of Redis instances" right? :-)

snc commented:
Thanks, merged.

othillo commented:
thanks! would you mind tagging a new release?

sschueller commented on Resources/doc/index.md:
```diff
@@ -120,6 +120,24 @@ snc_redis:
+            type: predis
+            alias: default
+            dsn:
+                - redis://localhost
```
Are these sentinel hosts or redis hosts? What about port? Does the `replication: sentinel` setting automatically use port `26379` on those hostnames?

ricbra commented:
Could you tag a new release? @snc

mnapier82 commented:
For anyone subscribed to this I see that a new release was tagged 8 days ago that includes this.
Thank you!
